src/query/service/src/pipelines/processors/transforms/window/transform_window.rs

```diff
@@ -109,6 +109,8 @@ pub struct TransformWindow<T: Number> {
     frame_end: RowPtr,
     frame_started: bool,
     frame_ended: bool,
+    // for percent_rank/cume_dist
+    frame_size: usize,
```

Comment:
We can use partition rows.

The definition in DuckDB:

```
The relative rank of the current row: (rank() - 1) / (total partition rows - 1).
```

Reply:
seems snowflake just only this in SQL but still use whole partitition for compute

Reply:
The frame definition will not affect the result. Just like `row_number` and `rank`.

Reply:
done.

I mean snowflake not allow ` range between 1 preceding and 1 following`,
while mysql and pg just allow any frame clause by ignore it.

Reply:
Ok.